llvm/lib/Transforms/Coroutines/CoroAnnotationElide.cpp

```diff
@@ -145,6 +191,37 @@ PreservedAnalyses CoroAnnotationElidePass::run(LazyCallGraph::SCC &C,
       bool IsCallerPresplitCoroutine = Caller->isPresplitCoroutine();
       bool HasAttr = CB->hasFnAttr(llvm::Attribute::CoroElideSafe);
       if (IsCallerPresplitCoroutine && HasAttr) {
+
+        llvm::raw_ostream *OS = nullptr;
+        auto _ = getCoroElidedStatsOStream(OS);
+        assert(OS && "At least we should able to get access to standard error");
+
+        auto &BFI = FAM.getResult<BlockFrequencyAnalysis>(*Caller);
+        if (BFI.getBlockFreq(CB->getParent()) <
+            BFI.getEntryFreq()) {
+          static BranchProbability MinBranchProbability(
+              static_cast<int>(CoroElideBranchRatio * MinBlockCounterExecution),
+              MinBlockCounterExecution);
+
+          auto Prob = BranchProbability::getBranchProbability(
+              BFI.getBlockFreq(CB->getParent()).getFrequency(),
+              BFI.getEntryFreq().getFrequency());
+
+          if (Prob < MinBranchProbability) {
+            *OS << "Not eliding " << *CB
+                << " with estimated probability: " << Prob << "\n";
+            continue;
+          }
+
+          *OS << "BB Prob: \t" << Prob << "\n";
+        } else {
+          *OS << "BB Freq: \t"
+              << BFI.getBlockFreq(CB->getParent()).getFrequency() << "\n";
+          *OS << "Entry Freq: \t" << BFI.getEntryFreq().getFrequency() << "\n";
+        }
+
+        *OS << "eliding " << *CB << "\n";
```

Comment:
should we integrate this information on whether / why a coroutine was elided into the optimization remarks (already used further down in this file)

Reply:
I think your PR might actually become simpler if you integrate with optimization remarks right away. You might be able to remove the debug settings and the custom logic to determine the output stream

Reply:
If you want it to make the 21 cut, why not remove the part that emits the stats and leave a FIXME for future optimization remarks work? Also, it really doesn't take that long.

Reply:
> The ORE variable is just 10 lines away from the hunk of your change.

Yeah, but if anything surprising happens, it may take me 30+ minutes to, maybe, one hour. I just don't want to have a risk for me now.

Reply:
> I think we should definitely still get this patch in before the 21.0 branch cut

I think if you want that, may be you can take it after I land this.